lib/actions/posts.ts

```diff
@@ -107,6 +107,62 @@ async function _getPost(id: string) {
   }
 }
 
+export async function deletePost(postID: string, userID: string) {
+  try {
+    await connectToDB();
+    const session = await getSession();
+    const sessionUser = session?.user;
+
+    if (!sessionUser) {
+      throw new Error("You must be logged in to delete a post");
+    }
+
+    if (userID !== sessionUser.id) {
+      throw new Error("You are not authorized to delete this post");
+    }
+
+    //delete the post from the database
+    await Note.deleteOne({ _id: postID });
+
+    //delete the post from the user's notes
+    await User.findOneAndUpdate({ _id: userID }, { $pull: { notes: postID } });
+    revalidateTag("posts");
+  } catch (err) {
+    console.log(err);
+  }
+}
```

Comment:
Not sure if I'm a fan of this pattern here. We're not really adding any security by passing in the creator ID to this function. This function is being called from the client- who could send anything the want anyway. There's no added security here beyond fetching the session.

I think it would make more sense to instead just send the noteID into this function- and then compare the session id to the note's creator to validate if they should be able to delete this or not.
I'll explain a bit more in a video below:

https://www.loom.com/share/79caeff269a949f7ac106c84f42379bd

Reply:
Hi Brandon, this makes a lot of sense -- thanks for walking through that with me!